crates/api_crud/src/site/update.rs

```diff
@@ -45,18 +44,24 @@ impl PerformCrud for EditSite {
     let local_site = site_view.local_site;
     let site = site_view.site;
 
-    // Make sure user is an admin
+    // BEGIN VALIDATION
+    // Make sure user is an admin; other types of users should not update site data...
     is_admin(&local_user_view)?;
 
-    let slur_regex = local_site_to_slur_regex(&local_site);
+    // Check that the slur regex compiles, and return the regex if valid...
+    let slur_regex = build_and_check_regex(&local_site.slur_filter_regex.as_deref())?;
 
-    check_slurs_opt(&data.name, &slur_regex)?;
-    check_slurs_opt(&data.description, &slur_regex)?;
+    if let Some(name) = &data.name {
+      site_name_length_check(name)?;
+      check_slurs_opt(&data.name, &slur_regex)?;
+    }
 
     if let Some(desc) = &data.description {
       site_description_length_check(desc)?;
+      check_slurs_opt(&data.description, &slur_regex)?;
```

Comment:
Same concern as above, and this also should probably be changed to `if let Some(Some(...` . 

I hope it won't fail on empty strings.

Reply:
I had started moving the validation portions into a function; based on this and your previous comments I'll just go ahead and do that! It'll let me write more targeted unit tests and show things should continue to work as expected 👍

Reply:
You can see this working as (I believe) expected here: https://github.com/LemmyNet/lemmy/pull/3146/files#diff-fcc9fa5b2820776716027bd57d2b8c83dfd11b52b3274a1cb341e5845aa1feebR370